### PR TITLE
[OTel C++] OTel plugin improvements

### PR DESCRIPTION
**Pull request by yashykt**
Changes -
* Use `grpc-c++` as the meter name and the proper version string when creating the meter for OTel
* setting metric description and unit. (Pointed out by @DNVindhya )

### CODE REVIEW

**Review by markdroth — APPROVED** [review]

File: src/cpp/ext/otel/otel_plugin.cc [hunk 1]
```diff
@@ -175,47 +176,61 @@ void OpenTelemetryPluginBuilder::BuildAndRegisterGlobal() {
...
   }
   if (metrics_.contains(
           OTelServerCallRcvdTotalCompressedMessageSizeInstrumentName())) {
     g_otel_plugin_state_->server.call.rcvd_total_compressed_message_size =
-        meter->CreateUInt64Histogram(std::string(
-            OTelServerCallRcvdTotalCompressedMessageSizeInstrumentName()));
+        meter->CreateUInt64Histogram(
+            std::string(
+                OTelServerCallRcvdTotalCompressedMessageSizeInstrumentName()),
+            "Compressed message bytes sent per server call", "By");
```
Comment (DNVindhya) [inline src/cpp/ext/otel/otel_plugin.cc:233]: s/sent/received

File: src/cpp/ext/otel/otel_plugin.cc [hunk 2]
```diff
@@ -175,47 +176,61 @@ void OpenTelemetryPluginBuilder::BuildAndRegisterGlobal() {
   if (meter_provider == nullptr) {
     return;
   }
-  auto meter = meter_provider->GetMeter("grpc");
+  auto meter = meter_provider->GetMeter("grpc-c++", GRPC_CPP_VERSION_STRING);
   if (metrics_.contains(OTelClientAttemptStartedInstrumentName())) {
     g_otel_plugin_state_->client.attempt.started = meter->CreateUInt64Counter(
-        std::string(OTelClientAttemptStartedInstrumentName()));
+        std::string(OTelClientAttemptStartedInstrumentName()),
+        "Number of call attempts started", "{attempt}");
```
Comment (DNVindhya) [inline src/cpp/ext/otel/otel_plugin.cc:183]: Nit: Can we use "client" in the description in the client metrics, if that makes sense?

**Review by DNVindhya — COMMENTED** [review]
LGTM

**Review by DNVindhya — APPROVED** [review]

File: src/cpp/ext/otel/otel_plugin.cc (hunk 1 above)
Comment (yashykt) [reply src/cpp/ext/otel/otel_plugin.cc]: Thanks! I'm so glad I asked you to take a look :D

File: src/cpp/ext/otel/otel_plugin.cc (hunk 2 above)
Comment (yashykt) [reply src/cpp/ext/otel/otel_plugin.cc]: Added

**Pull request merged**